py-polars/src/dataframe.rs

```diff
@@ -50,14 +50,15 @@ impl PyDataFrame {
     fn finish_from_rows(
         rows: Vec<Row>,
         infer_schema_length: Option<usize>,
-        schema_overwrite: Option<Schema>,
+        schema: Option<Schema>,
+        schema_overrides_by_idx: Option<PlHashMap<usize, DataType>>,
     ) -> PyResult<Self> {
         // Object builder must be registered, this is done on import.
-        let schema =
-            rows_to_schema_supertypes(&rows, infer_schema_length.map(|n| std::cmp::max(1, n)))
-                .map_err(PyPolarsErr::from)?;
+        let s = rows_to_schema_supertypes(&rows, infer_schema_length.map(|n| std::cmp::max(1, n)))
+            .map_err(PyPolarsErr::from)?;
+
         // Replace inferred nulls with boolean and erase scale from inferred decimals.
-        let fields = schema.iter_fields().map(|mut fld| match fld.data_type() {
+        let fields = s.iter_fields().map(|mut fld| match fld.data_type() {
```

Comment:
This code was already here, but I don't understand why we don't modify the data types in place. As we convert to `final_schema` yet again on line 73.